pkg/source/ova/client.go

```diff
@@ -0,0 +1,537 @@
+func (c *Client) GenerateVirtualMachine(vmi *migration.VirtualMachineImport) (*kubevirtv1.VirtualMachine, error) {
+	tempArchivePath := c.generateArchivePath(vmi)
+
+	e, err := readEnvelope(tempArchivePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read envelope: %w", err)
+	}
+
+	fw, hw, ni := parseEnvelope(e, vmi.GetDefaultNetworkInterfaceModel())
+	logrus.WithFields(util.FieldsToJSON(logrus.Fields{
+		"name":         vmi.Name,
+		"namespace":    vmi.Namespace,
+		"firmware":     fw,
+		"hardware":     hw,
+		"networkInfos": ni,
+	}, []string{"firmware", "hardware", "networkInfos"})).Info("Parsed settings from OVF envelope")
+
+	newVM := &kubevirtv1.VirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      vmi.Status.ImportedVirtualMachineName,
+			Namespace: vmi.Namespace,
+		},
+	}
+
+	vmSpec := source.NewVirtualMachineSpec(source.VirtualMachineSpecConfig{
+		Name: vmi.Status.ImportedVirtualMachineName,
+		Hardware: source.Hardware{
+			NumCPU:            hw.NumCPU,
+			NumCoresPerSocket: 1, // OVF does not provide cores per socket information.
+			MemoryMB:          hw.MemoryMB,
+		},
+	})
+
+	mappedNetwork := source.MapNetworks(ni, vmi.Spec.Mapping)
+	networkConfig, interfaceConfig := source.GenerateNetworkInterfaceConfigs(mappedNetwork, vmi.GetDefaultNetworkInterfaceModel())
+
+	// Setup BIOS/EFI, SecureBoot and TPM settings.
+	source.ApplyFirmwareSettings(vmSpec, fw)
+
+	vmSpec.Template.Spec.Networks = networkConfig
+	vmSpec.Template.Spec.Domain.Devices.Interfaces = interfaceConfig
+	newVM.Spec = *vmSpec
+
+	return newVM, nil
+}
+
+// ShutdownGuest is required by the `VirtualMachineOperations` interface.
+func (c *Client) ShutdownGuest(_ *migration.VirtualMachineImport) error {
+	// Nothing to do here.
+	return nil
+}
+
+// PowerOff is required by the `VirtualMachineOperations` interface.
+func (c *Client) PowerOff(_ *migration.VirtualMachineImport) error {
+	// Not implemented as OVA does not support guest OS operations.
+	return nil
+}
+
+// IsPowerOffSupported is required by the `VirtualMachineOperations` interface.
+func (c *Client) IsPowerOffSupported() bool {
+	// Powering off the VM is not supported.
+	return false
+}
+
+// IsPoweredOff is required by the `VirtualMachineOperations` interface.
+func (c *Client) IsPoweredOff(_ *migration.VirtualMachineImport) (bool, error) {
+	// The VM is always considered powered off.
+	return true, nil
+}
+
+func (c *Client) Cleanup(vmi *migration.VirtualMachineImport) error {
+	_ = os.Remove(c.generateArchivePath(vmi))
+	return source.RemoveTempImageFiles(vmi.Status.DiskImportStatus)
+}
+
+// downloadArchive downloads the OVA file to /tmp.
+func downloadArchive(srcPath, dstPath string, secret *corev1.Secret) error {
+	logrus.WithFields(logrus.Fields{
+		"srcPath":      srcPath,
+		"dstPath":      dstPath,
+		"useBasicAuth": secret != nil,
+	}).Info("Downloading OVA archive")
+
+	httpClient := &http.Client{
```

Comment:
we should probably store the http client in the Client object so it can be used across different operations.

We should also add ability to provide custom ca cert via the secret reference. with the http client being stored as part of the OVA client, it will make it easier to initialise this once and reuse across the entire workflow.